src/nrfx/hal/nrf_radio.h

```diff
@@ -140,6 +508,215 @@ void nrf_radio_bcc_set(NRF_RADIO_Type * p_reg, uint32_t radio_bcc);
+/*****************************/
+
+NRF_STATIC_INLINE void nrf_radio_event_clear(NRF_RADIO_Type * p_reg, nrf_radio_event_t event)
+{
+    *((volatile uint32_t *)((uint8_t *)p_reg + (uint32_t)event)) = 0x0UL;
+#if __CORTEX_M == 0x04
+    volatile uint32_t dummy = *((volatile uint32_t *)((uint8_t *)p_reg + (uint32_t)event));
+    (void)dummy;
+#endif
+}
+
+NRF_STATIC_INLINE bool nrf_radio_event_check(NRF_RADIO_Type const * p_reg, nrf_radio_event_t event)
+{
+    return (bool) *((volatile uint32_t *)((uint8_t *)p_reg + (uint32_t)event));
+}
+
+NRF_STATIC_INLINE void nrf_radio_shorts_enable(NRF_RADIO_Type * p_reg, uint32_t shorts_mask)
+{
+    p_reg->SHORTS |= shorts_mask;
+}
+
+NRF_STATIC_INLINE void nrf_radio_shorts_disable(NRF_RADIO_Type * p_reg, uint32_t shorts_mask)
+{
+    p_reg->SHORTS &= ~shorts_mask;
+}
+
+NRF_STATIC_INLINE void nrf_radio_shorts_set(NRF_RADIO_Type * p_reg, uint32_t shorts_mask)
+{
+    p_reg->SHORTS = shorts_mask;
+}
+
+NRF_STATIC_INLINE uint32_t nrf_radio_int_enable_check(NRF_RADIO_Type const * p_reg, uint32_t mask)
+{
+    return p_reg->INTENSET & mask;
+}
+
+NRF_STATIC_INLINE bool nrf_radio_crc_status_check(NRF_RADIO_Type const * p_reg)
+{
+    return ((p_reg->CRCSTATUS & RADIO_CRCSTATUS_CRCSTATUS_Msk) >> RADIO_CRCSTATUS_CRCSTATUS_Pos)
+             == RADIO_CRCSTATUS_CRCSTATUS_CRCOk ;
+}
+
+NRF_STATIC_INLINE uint8_t nrf_radio_dai_get(NRF_RADIO_Type const * p_reg)
+{
+    return (uint8_t)p_reg->DAI;
+}
+
+NRF_STATIC_INLINE void nrf_radio_packetptr_set(NRF_RADIO_Type * p_reg, void const * p_packet)
+{
+    p_reg->PACKETPTR = (uint32_t)p_packet;
+}
+
+NRF_STATIC_INLINE void * nrf_radio_packetptr_get(NRF_RADIO_Type const * p_reg)
+{
+    return (void *)p_reg->PACKETPTR;
+}
+
+NRF_STATIC_INLINE void nrf_radio_frequency_set(NRF_RADIO_Type * p_reg, uint16_t radio_frequency)
+{
+    NRFX_ASSERT(radio_frequency <= 2500);
+    NRFX_ASSERT(radio_frequency >= 2400);
+    p_reg->FREQUENCY = (uint32_t)(radio_frequency - 2400);
+}
+
+NRF_STATIC_INLINE void nrf_radio_txpower_set(NRF_RADIO_Type * p_reg, nrf_radio_txpower_t tx_power)
+{
+    p_reg->TXPOWER = (((uint32_t)tx_power) << RADIO_TXPOWER_TXPOWER_Pos);
+}
+
+NRF_STATIC_INLINE void nrf_radio_mode_set(NRF_RADIO_Type * p_reg, nrf_radio_mode_t radio_mode)
+{
+    p_reg->MODE = ((uint32_t) radio_mode << RADIO_MODE_MODE_Pos);
+}
+
+NRF_STATIC_INLINE nrf_radio_mode_t nrf_radio_mode_get(NRF_RADIO_Type const * p_reg)
+{
+    return (nrf_radio_mode_t)((p_reg->MODE & RADIO_MODE_MODE_Msk) >> RADIO_MODE_MODE_Pos);
+}
+
+NRF_STATIC_INLINE void nrf_radio_packet_configure(NRF_RADIO_Type *                p_reg,
+                                                  nrf_radio_packet_conf_t const * p_config)
+{
+    p_reg->PCNF0 = (((uint32_t)p_config->lflen << RADIO_PCNF0_LFLEN_Pos) |
+                    ((uint32_t)p_config->s0len << RADIO_PCNF0_S0LEN_Pos) |
+                    ((uint32_t)p_config->s1len << RADIO_PCNF0_S1LEN_Pos) |
+                    (p_config->s1incl ?
```

Comment:
Same question about not including the reset of the code here
(even if it those defines are 1 or 0 in the HW models, and therefore the code would be in practice the same, minimizing differences in the code should help maintain these files in the future)